src/Content/Pager.php

```diff
@@ -0,0 +1,282 @@
+<?php
+
+namespace Friendica\Content;
+
+use Friendica\Core\L10n;
+
+/**
+ * The Pager has two very different output, Minimal and Full, see renderMinimal() and renderFull() for more details.
+ *
+ * @author Hypolite Petovan <mrpetovan@gmail.com>
+ */
+class Pager
+{
+	/**
+	 * @var integer
+	 */
+	private $page = 1;
+	/**
+	 * @var integer
+	 */
+	private $itemsPerPage = 50;
+
+	/**
+	 * @var string
+	 */
+	private $baseQueryString = '';
+
+	/**
+	 * Instantiates a new Pager with the base parameters.
+	 *
+	 * Guesses the page number from the GET parameter 'page'.
+	 *
+	 * @param string  $queryString  The query string of the current page
+	 * @param integer $itemsPerPage An optional number of items per page to override the default value
+	 */
+	public function __construct($queryString, $itemsPerPage = 50)
+	{
+		$this->setQueryString($queryString);
+		$this->setItemsPerPage($itemsPerPage);
+		$this->setPage(defaults($_GET, 'page', 1));
+	}
+
+	/**
+	 * Returns the start offset for a LIMIT clause. Starts at 0.
+	 *
+	 * @return integer
+	 */
+	public function getStart()
+	{
+		return max(0, ($this->page * $this->itemsPerPage) - $this->itemsPerPage);
+	}
+
+	/**
+	 * Returns the number of items per page
+	 *
+	 * @return integer
+	 */
+	public function getItemsPerPage()
+	{
+		return $this->itemsPerPage;
+	}
+
+	/**
+	 * Returns the current page number
+	 *
+	 * @return type
+	 */
+	public function getPage()
+	{
+		return $this->page;
+	}
+
+	/**
+	 * Returns the base query string.
+	 *
+	 * Warning: this isn't the same value as passed to the constructor.
+	 * See setQueryString() for the inventory of transformations
+	 *
+	 * @see setBaseQuery()
+	 * @return string
+	 */
+	public function getBaseQueryString()
+	{
+		return $this->baseQueryString;
+	}
+
+	/**
+	 * Sets the number of items per page, 1 minimum.
+	 *
+	 * @param integer $itemsPerPage
+	 */
+	public function setItemsPerPage($itemsPerPage)
+	{
+		$this->itemsPerPage = max(1, intval($itemsPerPage));
+	}
+
+	/**
+	 * Sets the current page number. Starts at 1.
+	 *
+	 * @param integer $page
+	 */
+	public function setPage($page)
+	{
+		$this->page = max(1, intval($page));
+	}
+
+	/**
+	 * Sets the base query string from a full query string.
+	 *
+	 * Strips the 'page' parameter, and remove the 'q=' string for some reason.
+	 *
+	 * @param string $queryString
+	 */
+	public function setQueryString($queryString)
```

Comment:
Just a suggestion :-)

```php
function testQueryString() 
{
       $pager = new Paget(..);
       $queryString = 'a/query/string?key=value&q=value';
       $stripped = 'a/query/string?key=value';
       $pager->setQueryString($queryString);
       $this->assertSame($stripped, $this->getBaseQueryString());
}
```

Reply:
I'll add this test in a subsequent PR. Thank you for the suggestion!